### PR TITLE
Lesson 11

### PR DESCRIPTION
**Pull request by ArtemNeT13**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -75,15 +75,49 @@ <h1>Discover stays<br>
                     </div>
                     <div class="filter-block col-sm-12">
                         <div class="filter-label">
-                            <label>2 Adults — 0 Children — 1 Room</label>
+                            <label  id="lable_value">2 Adults — 0 Children — 1 Room</label>
```
Comment (bogomolov-and) [inline index.html]: Remove extra space before `id`, please

File: index.html [hunk 2]
```diff
@@ -75,15 +75,49 @@ <h1>Discover stays<br>
...
                         </div>
                         <div class="filter-input">
-                            <input type="text" name="filter-abults" placeholder="2 Adults">
+                            <input id="filter-input__adults" type="text" name="filter-adults" value="2 Adults">
```
Comment (bogomolov-and) [inline index.html]: All id's can be named without BEM. Use simple kebab-case. For example, `filter-adults`

File: main.js [hunk 3]
```diff
@@ -1,88 +1,217 @@
...
+//         imageUrl: 'https://res.cloudinary.com/intellectfox/image/upload/v1610379365/fe/villa-kunerad_gdbqgv.jpg',
+//     },
+//     {
+//         name: 'Hostel Friendship',
+//         city: 'Berlin',
+//         country: 'Germany',
+//         imageUrl: 'https://res.cloudinary.com/intellectfox/image/upload/v1610379364/fe/hostel-friendship_aw6tn7.jpg',
+//     },
+//     {
+//         name: 'Radisson Blu Hotel',
```
Comment (bogomolov-and) [inline main.js]: Such comments should be removed. If you need previous versions, you can use git for that :)

File: main.js [hunk 4]
```diff
@@ -1,88 +1,217 @@
...
+let adultsValue = document.getElementById("adults_value");
+let adultsVal = parseFloat(adultsValue.value);
+let adultsColorMinus = document.getElementById("counter-adults__minus");
+let adultsColorPlus = document.getElementById("counter-adults__plus");
+let adultsInput = document.getElementById("filter-input__adults");
+document.getElementById("counter-adults__plus").onclick = function() {
+    if (adultsVal < maxAdults) {
+        adultsValue.value = ++adultsVal;
+        adultsColorMinus.style.color = "#3077C6";
+        adultsColorMinus.style.borderColor = "#3077C6";
```
Comment (bogomolov-and) [inline main.js]: Such styles should be moved to css file. Use `classList` for that

File: main.js [hunk 5]
```diff
@@ -1,88 +1,217 @@
...
+        const dataNew = data.slice(0, 4)
+        let hotelOffer = ''
+        let favoriteHotels = dataNew.reduce((prev, hotel) => {
+            return hotelOffer += `<div class="hotel-foto foto-on col-xl-3 col-sm-6">
                 <div class="hotel-img col-sm-12">
                    <img src="${hotel.imageUrl}" alt="picture">
                 </div>
                 <p>${hotel.name}</p>
                 <p><span>${hotel.city}, ${hotel.country}</span></p>
             </div>`
```
Comment (bogomolov-and) [inline main.js]: Look, you have `prev` as an empty string and you have `hotelOffer` as an empty string. I think you can use one :)

File: main.js [hunk 6]
```diff
@@ -1,88 +1,217 @@
...
+        roomsColorMinus.style.color = "#CECECE";
+        roomsColorMinus.style.borderColor = "#CECECE";
+    }
+    roomsInput.value = `${roomsVal} Rooms`;
+    lableValue.innerHTML  = `${adultsVal} Adults — ${childrenVal} Children — ${roomsVal} Room`;
+}
+
+//                  LESSON-12
+let url = 'https://fe-student-api.herokuapp.com/api/hotels/popular'
+fetch(url)
```
Comment (bogomolov-and) [inline main.js]: Do you know how can use `fetch` with `await`?
Spoiler, you can use self-invoking function as an async function :)

File: main.js [hunk 7]
```diff
@@ -162,39 +165,73 @@ btnSearchHeader.addEventListener('click', (event) => {
...
-                return prev += `<div class="hotel-foto foto-on col-xl-3 col-sm-6">
+
+    fetch(urlGet)
+        .then((response) => {
+            return response.json()
+                .then((data) => {
+                    const hotelData = JSON.stringify(data);
+                    sessionStorage.setItem('ajaxRequest', hotelData)
+                })
+        })
```
Comment (romos21) [inline main.js:189]: why then is into then?

File: main.js [hunk 8]
```diff
@@ -162,39 +165,73 @@ btnSearchHeader.addEventListener('click', (event) => {
...
+//                        <img src="${hotel.imageUrl}" alt="picture">
+//                     </div>
+//                     <p>${hotel.name}</p>
+//                     <p><span>${hotel.city}, ${hotel.country}</span></p>
+//                     </div>`
+//             }, '')
+//             const searchDiv = document.getElementById("searchResult")
+//             searchDiv.innerHTML = searchHotels
+//         })
+// })
```
Comment (romos21) [inline main.js:235]: remove comments, it makes code unreadable

File: main.js [hunk 9]
```diff
@@ -162,39 +165,73 @@ btnSearchHeader.addEventListener('click', (event) => {
...
-        }else{
+        } else {
             children = ''
-            for(let value of selectAge.querySelectorAll("select")){
+            for (let value of selectAge.querySelectorAll("select")) {
                 children += `${value.value},`
             }
         }
     }
-    fetch(`${urlGet}?search=${search}&adults=${adults}&children=${children}&rooms=${rooms}`)
```
Comment (romos21) [inline main.js]: Why did you remove this URL string? I remember we did that together

File: main.js [hunk 10]
```diff
@@ -155,70 +155,34 @@ document.getElementById("counter-rooms-minus").onclick = () => {
...
-
 btnSearchHeader.addEventListener('click', (event) => {
     event.preventDefault()
-    fetch(urlGet)
+    let search = document.getElementById('searchHeader').value
+    let adults = document.getElementById('adults-value').value
+    let children = document.getElementById('children-value').value
+    let rooms = document.getElementById('rooms-value').value
+
+    let isValid = false;
```
Comment (romos21) [inline main.js]: So, if you use "return" when validation failed you can remove this variable. The reason was that we needed to block fetch it validation failed

File: main.js [hunk 11]
```diff
@@ -162,39 +165,73 @@ btnSearchHeader.addEventListener('click', (event) => {
...
+            for (let value of selectAge.querySelectorAll("select")) {
                 children += `${value.value},`
             }
         }
     }
-    fetch(`${urlGet}?search=${search}&adults=${adults}&children=${children}&rooms=${rooms}`)
-        .then((response) => response.json())
-        .then((dataGet) => {
-            const availableHotels = document.getElementById("availableHotels")
-            availableHotels.style.display = "flex";
```
Comment (romos21) [inline main.js]: Why did you remove that?

File: main.js [hunk 12]
```diff
@@ -162,39 +165,73 @@ btnSearchHeader.addEventListener('click', (event) => {
...
-            let searchHotels = dataGetNew.reduce((prev, hotel) => {
-                return prev += `<div class="hotel-foto foto-on col-xl-3 col-sm-6">
+
+    fetch(urlGet)
+        .then((response) => {
+            return response.json()
+                .then((data) => {
+                    const hotelData = JSON.stringify(data);
+                    sessionStorage.setItem('ajaxRequest', hotelData)
+                })
```
Comment (romos21) [inline main.js]: why then is into then?

File: main.js (hunk 8 above)
Comment (romos21) [inline main.js]: dont paste long comments - it makes your code unreadable

File: main.js [hunk 13]
```diff
@@ -179,59 +178,57 @@ btnSearchHeader.addEventListener('click', (event) => {
...
+                <p><span>${hotel.city}, ${hotel.country}</span></p>
+            </div>`
+    },'')
+    const currentDiv = document.getElementById("homeGuests")
+    currentDiv.innerHTML = favoriteHotels
+}
+if (requestToStorage) {
+    const requestToStorageParse = JSON.parse(requestToStorage)
+    requestDataHotels(requestToStorageParse)
+}else{
```
Comment (romos21) [inline main.js:238]: think about code formating

**Comment by romos21** [comment]
@ArtemNeT13 @bogomolov-and LGTM!

**Review by bogomolov-and — APPROVED** [review]
